packages/cardano-services-client/src/WebSocket.ts

```diff
@@ -0,0 +1,208 @@
+/* eslint-disable unicorn/prefer-add-event-listener */
+// cSpell:ignore njson
+
+import { AsyncReturnType, Cardano, NetworkInfoProvider, RequestMethods, WSMessage } from '@cardano-sdk/core';
+import { Logger } from 'ts-log';
+import { Observable, ReplaySubject, firstValueFrom } from 'rxjs';
+
+import { NJSON } from 'next-json';
+import WebSocket from 'isomorphic-ws';
+
+type WSStatus = 'connecting' | 'connected' | 'idle' | 'stop';
+
+export type WSHandler = (message: WSMessage) => void;
+type WSPerform = () => boolean;
+
+export interface WsClientConfiguration {
+  /** The interval in seconds between two heartbeat messages. Default 55". */
+  heartbeatInterval?: number;
+
+  /** The interval in seconds after which a request must timeout. Default 60". */
+  requestTimeout?: number;
+
+  /** The WebSocket server URL. */
+  url: URL;
+}
+
+export interface WsClientDependencies {
+  /** The logger. */
+  logger: Logger;
+}
+
+const isEventError = (error: unknown): error is { error: Error } =>
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  typeof error === 'object' && !!error && (error as any).error instanceof Error;
+
+export class CardanoWsClient {
+  /** The client id, assigned by the server. */
+  clientId = 'not-connected';
```

Comment:
Not for the POC. The `not-connected` string should be a constant.

Reply:
70f3fe7bc1 should solve this